pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -149,80 +149,38 @@ func (r *RepoSyncReconciler) Reconcile(ctx context.Context, req controllerruntim
 		return controllerruntime.Result{}, status.APIServerError(err, "failed to get RepoSync")
 	}
 
-	if !rs.DeletionTimestamp.IsZero() {
-		// Deletion requested.
-		// Cleanup is handled above, after the RepoSync is NotFound.
-		r.logger(ctx).V(3).Info("Sync deletion timestamp detected")
-	}
-
 	currentRS := rs.DeepCopy()
 
-	if err := r.watchConfigMaps(rs); err != nil {
-		r.logger(ctx).Error(err, "Error watching ConfigMaps")
-		reposync.SetStalled(rs, "ConfigMapWatch", err)
-		_, updateErr := r.updateStatus(ctx, currentRS, rs)
-		if updateErr != nil {
-			r.logger(ctx).Error(updateErr, "Error setting status")
+	if rs.DeletionTimestamp.IsZero() {
+		// Only validate RepoSync if it is not deleting. Otherwise, the validation
+		// error will block the finalizer.
+		if err := r.watchConfigMaps(rs); err != nil {
+			r.logger(ctx).Error(err, "Error watching ConfigMaps")
+			reposync.SetStalled(rs, "ConfigMapWatch", err)
+			_, updateErr := r.updateStatus(ctx, currentRS, rs)
+			if updateErr != nil {
+				r.logger(ctx).Error(updateErr, "Error setting status")
+			}
+			// the error may be recoverable, so we retry (return the error)
+			return controllerruntime.Result{}, updateErr
```

Comment:
i think we want to return the `err` here, not the `updateErr` that came afterwards. That's why we log the  `updateErr`.

Reply:
Thanks for capturing it. Done.